examples/ferris.rs

```diff
@@ -0,0 +1,535 @@
+//! Draw the officially unofficial Rust mascot.
+//!
+//! # Synopsis
+//!
+//! This example draws a happy Ferris, the Rust language mascot. The original
+//! image was taken from the Rust community website [rustacean.net].
+//!
+//! # Implementation details
+//!
+//! As the image is quite curvy, building arcs by hand would have been awful.
+//! So this example uses a [cubic Bézier curve] implementation extensively.
+//! It does it by first using [GIMP] in order to measure each curve's cubic
+//! parameters and simply paste them in the drawing code. Those are "unadapted"
+//! points because the original and actual drawing aspect ratios are not
+//! necessarily the same. So they are first "adapted" by distorting the space.
+//! They are also absolute points because using them directly would not take
+//! into account the current position of the turtle. So they are then transformed
+//! into relative points. Finally, the current position is added into the mix in
+//! order to get absolute points back and draw the cubic Bézier curve simply by
+//! applying the formula. Some linear movements are of course used as well.
+//!
+//! As one can see, there has been a great deal of effort put into making the
+//! drawing as independent as possible of the canvas size. Although it worked
+//! fine for the first parts of the drawing, it does not for the shell spikes
+//! unfortunately. This is because measuring curve parameters takes a long time,
+//! so to avoid that, a simple loop is used to draw the spikes. For them to be
+//! drawn correctly however, a rotation of the Bézier points had to be implemented,
+//! otherwise the crab would be flat. The ratio adaptation could not be added
+//! in the rotation, so resizing the canvas results in chaos sadly.
+//!
+//! Solutions to fix this could be:
+//!  * Finding a way to adapt rotated adapted relative points.
+//!  * Using elliptic arcs for the spike tip parts, which would be convenient
+//!    to adapt.
+//!  * Removing all the ratio adaptation, replacing it with size adaptation and
+//!    using a canvas size with the same aspect ratio as the original image.
+//!
+//! Any contribution or suggestion towards this is welcome!
+//!
+//! [rustacean.net]: https://rustacean.net/
+//! [cubic Bézier curve]: https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Cubic_B%C3%A9zier_curves
+//! [GIMP]: https://www.gimp.org/
+
+/// Imports of most of `turtle`'s API.
+use turtle::{Angle, Color, Distance, Drawing, Point, Size, Turtle};
+
+/// Provides a rotation method producing new values.
+trait Rotate {
+    /// Rotates in the 2D space by the given angle into a new value.
+    fn rot(&self, angle: Angle) -> Self;
+}
+
+/// Extension of the `Point` struct.
+impl Rotate for Point {
+    /// 2D counterclockwise rotates the point by the given angle.
+    fn rot(&self, angle: Angle) -> Self {
```

Comment:
It's probably a good idea to document that this assumes that `Angle` is in radians (since by default it is in degrees)

Reply:
I added appropriate comments.